### PR TITLE
added generic DSL support using metadata from Descriptors

### PR DESCRIPTION
**Pull request by daspilker**

### CODE REVIEW

**Comment by daspilker** [comment]
I have verified that the following plugins work with generic DSL support:

* [Image Gallery Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Image+Gallery+Plugin) ([example](https://github.com/jenkinsci/job-dsl-plugin/pull/728#issuecomment-207344164))
* [Gerrit Trigger](https://wiki.jenkins-ci.org/display/JENKINS/Gerrit+Trigger)
* [CVS Plugin](https://wiki.jenkins-ci.org/display/JENKINS/CVS+Plugin)
* [Unity3dBuilder Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Unity3dBuilder+Plugin) ([example](https://groups.google.com/d/msg/job-dsl-plugin/0SHN59GnW4A/a8ia4DERBAAJ))
* [Log Parser Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Log+Parser+Plugin) ([example](https://groups.google.com/d/msg/job-dsl-plugin/0SHN59GnW4A/a8ia4DERBAAJ))
* [Phing Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Phing+Plugin) ([example](https://groups.google.com/d/msg/job-dsl-plugin/0SHN59GnW4A/a8ia4DERBAAJ))
* [Job Exporter Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Job+Exporter+Plugin) ([example](https://groups.google.com/d/msg/job-dsl-plugin/0SHN59GnW4A/a8ia4DERBAAJ))
* [MSBuild Plugin](https://wiki.jenkins-ci.org/display/JENKINS/MSBuild+Plugin) ([example](https://groups.google.com/d/msg/job-dsl-plugin/0SHN59GnW4A/a8ia4DERBAAJ))
* [CloudBees Docker Hub Notification](https://wiki.jenkins-ci.org/display/JENKINS/CloudBees+Docker+Hub+Notification)
* [Version Number Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Version+Number+Plugin)

**Comment by v1v** [comment]
:+1:

File: gradle.properties [hunk 1]
```diff
@@ -1,3 +1,3 @@
 version=1.46-SNAPSHOT
 groovyVersion=1.8.9
-githubUser=jenkinsci
+githubUser=daspilker
```
Comment (sschuberth) [inline gradle.properties]: This change should probably be reverted, or?

File: Vagrantfile [hunk 2]
```diff
@@ -22,6 +22,10 @@ Vagrant.configure(2) do |config|
         master.vm.network "forwarded_port", guest: 8080, host: 8081
         master.vm.provision "shell", inline: $base_script
         master.vm.provision "shell", inline: $script
+
+        master.vm.provider "virtualbox" do |provider|
+            provider.memory = 640
```
Comment (sschuberth) [inline Vagrantfile]: Why does this become necessary? This change should probably be split out into its own commit, providing a proper reasoning in the commit message.

**Comment by daspilker** [comment]
@sschuberth thanks for the review, I removed those files from the commit

**Comment by chbloemer** [comment]
@daspilker is it possible to use the performance plugin with generic dsl? See: https://wiki.jenkins-ci.org/display/JENKINS/Performance+Plugin

**Comment by daspilker** [comment]
@chbloemer yes, it's possible

**Comment by itavero** [comment]
Is this included in release 1.46?

**Comment by sschuberth** [comment]
@itavero According to the [release notes](https://github.com/jenkinsci/job-dsl-plugin/wiki#release-notes), yes.

**Comment by itavero** [comment]
So, I got a hunch why this does not work. As mentioned in the comment I added to JENKINS-22138, I'm running the script locally.
I reckon I need to add the plug-in to my classpath or something, right?

**Comment by itavero** [comment]
Totally missed this page, [Automatically Generated DSL|https://github.com/jenkinsci/job-dsl-plugin/wiki/Automatically-Generated-DSL], which mentions some limitations in the last few paragraphs.
What I'm trying to achieve is not yet possible. :wink:
